### PR TITLE
feat(commands/issue): add EXAMPLES and ARGUMENTS

### PR DESCRIPTION
**Pull request by maxice8**
**Description**

This adds better messages to the issue and mr commands

**How Has This Been Tested?**

`bin/glab issue --help`

### CODE REVIEW

**Comment by maxice8** [comment]
@profclems I always wondered why we didn't have the very helpful `ARGUMENTS` message informing the user one can use `OWNER:branch` or `URL` instead of the `INTERNAL-ID` of the merge request.

image

File: commands/issue/issue.go [hunk 1]
```diff
@@ -22,6 +23,19 @@ func NewCmdIssue(f *cmdutils.Factory) *cobra.Command {
...
+			$ glab issue list
+			$ glab issue create --label --confidential
+			$ glab issue view --web
+			$ glab issue note -m "closing because !123 was merged" <issue number>
+		`),
+		Annotations: map[string]string{
+			"help:arguments": heredoc.Doc(`
+				An issue can be supplied as argument in any of the following formats:
+				- by number, e.g. "123"; or
+				- by URL, e.g. "https://gitlab.com/OWNER/REPO/-/issues/123".
```
Comment (profclems) [inline commands/issue/issue.go]: I don't think `glab` supports specifying issues by URL yet. Do you think implementing that?

Comment (maxice8) [reply commands/issue/issue.go]: I actually just assumed it did, I'll wait until somebody asks then I can look into implementing it

File: commands/mr/mr.go [hunk 2]
```diff
@@ -31,6 +32,18 @@ func NewCmdMR(f *cmdutils.Factory) *cobra.Command {
...
+		Example: heredoc.Doc(`
+			$ glab mr create --autofill --labels bugfix
+			$ glab mr merge 123
+			$ glab mr note -m "needs to do X before it can be merged" branch-foo
+		`),
+		Annotations: map[string]string{
+			"help:arguments": heredoc.Doc(`
+			A merge request can be supplied as argument in any of the following formats:
+			- by number, e.g. "123"; or
+			- by the name of its head branch, e.g. "patch-1" or "OWNER:patch-1".
```
Comment (profclems) [inline commands/mr/mr.go]: use `source branch` to make it much specific to GitLab:
```suggestion
			- by the name of its source branch, e.g. "patch-1" or "OWNER:patch-1".
```
or better
```suggestion
			- by the name of its head or source branch, e.g. "patch-1" or "OWNER:patch-1".
```

**Review by profclems — CHANGES_REQUESTED** [review]

File: commands/mr/mr.go (hunk 2 above)
Comment (maxice8) [reply commands/mr/mr.go]: fixed, thanks

**Review by profclems — APPROVED** [review]
Thanks a lot!
I didn't think about this

**Pull request merged**